src/context/AudioContext.js

```diff
@@ -0,0 +1,24 @@
+import * as React from 'react';
+
+const defaultErrorThunk = () => async () => {
```

Comment:
What happens here? Why this is a function that is evaluated so many times below?

Reply:
We could extend it to provide different conditions. Goes along with your over-engineered talk. I'll reuse the same function for each.